Add all svg and refactor all image names, also improve some

dpizetta opened the pull request:
- [X] Update all svg files and generate new images;
- [X] Update all paths and names;
- [x] Review todos - order reversed;
- [X] Fix tree view branches;
- [X] Improve window actions icons, close, minimize, undock;
- [X] Enhance dock window actions;
- [X] Fix tab title tab borders, close icon;
- [X] Revise all widgets and paths;
- [X] Check scripts that generate generic images;
- [x] Add image in the contribute guide + more guidelines;
- [x] Create a checker for matching names and quantities of png, svg and links in the files - almost done.

dpizetta commented:
@goanpeca could you take a look in the result? I'm in progress. Tks

goanpeca commented:
Awesome work @dpizetta ! So far looks good. Ping me when finished so I can take a another look!

goanpeca commented on CONTRIBUTING.md:
```diff
@@ -223,14 +238,27 @@ For more information about those scripts, see their documentation.
+
+- Size: 32px X 32px;
+- Border: 2px border space;
+- Line: 2px minimum thickness. Complementary 4px and 8px thickness;
```
I would add also that these lines and shape should align to the grid as much as possible (depending on the shape of course)

goanpeca replied:
Maybe adding an image?

https://hig.kde.org/style/icon.html
<img width="534" alt="Screen Shot 2019-05-24 at 17 10 50" src="https://user-images.githubusercontent.com/3627835/58358995-eec1ce80-7e46-11e9-9358-31a3f4efe87b.png">

dpizetta replied:
Oh, yes, excellent! I will add.

goanpeca commented:
We need to add a blacklist of svg, that should not be converted. `palette.svg` is one of them

goanpeca commented:
The arrows on the `File Explorer` of spyder look too big. Maybe  those files need a different size (not squared?)

<img width="294" alt="Screen Shot 2019-05-24 at 20 06 16" src="https://user-images.githubusercontent.com/3627835/58362684-7adff000-7e5f-11e9-80d1-76bd3ca2fb44.png">

goanpeca commented:
The close (used for tabs in spyder) looks small and not very readable

<img width="333" alt="Screen Shot 2019-05-24 at 20 47 41" src="https://user-images.githubusercontent.com/3627835/58363135-4d964080-7e65-11e9-9aae-6c38d0cf10a2.png">

dpizetta commented:
> We need to add a blacklist of svg, that should not be converted. palette.svg is one of them

Yep! I'll provide! tks

> The arrows on the File Explorer of spyder look too big. Maybe those files need a different size (not squared?)

Actually we may don't need to keep the size changes in the file itself, but I need to change in the scss file. I have also changed many ones, but I unnoticed this one. Also, I have added the lines to the treeview that was there but not used before. However, it seems too full, may thinner lines. I noticed that people aren't using the lines now-a-days. I will print both examples so we can decide.

> The close (used for tabs in spyder) looks small and not very readable.

Could you check if there is any special piece of code that change the style in the spyder application? As we had the different sizes, people could be used some code adjust in style, that now is not needed. For example, the close-tab in the 'example' seems to be in a nice size.

>It should probably be a filled icon, and have the x be the transparent part (like circular ones were). 

I'm using the same for the close window and dock widget. If we remove the box around the X and make it thicker. Something like this one. 

![image](https://user-images.githubusercontent.com/5084939/58370465-398e2580-7edd-11e9-9a3c-9b7affda1a5f.png)

I'm not a fan of keep the circle, because it leaves the interface heavier, but this is my opinion. 
@goanpeca  thanks for all the comments.

dpizetta commented:
@goanpeca finished! Could you take another look? Tks

goanpeca commented:
Yes! Will do in the afternoon
